### DISCUSSION TITLE
AppTheme bindings For Dark/light modes

### DISCUSSION THREAD

**Discussion by roughiain** [Dreamescaper/BlazorBindings.Maui#143] [Q&A] [answered]
What would be the best way to setup the styles to implement Dark/light mode.

**Comment by Dreamescaper** [comment, marked as answer]
There are multiple options. 

1. Set style directly in razor (e.g. AppTheme.Light ? Colors.White : Colors.Black). Not really pretty, but might work for the simplest apps.

2. Build a resource dictionary with C#, and apply it to your application globally or per control.

3. Build the same resource dictionary with XAML. I don't really like xaml for layout trees (hence this project),  but for styles only - why not.

4. You can also use css stylesheet, but I don't advise it - it's quite limited, and not really supported by MAUI devs anymore.


Those options have their difficulties, so I'll try to create a sample demonstrating them soonish .

**Reply by roughiain** [reply]
1 Does seem more like the Blazor way. It is a pity the CSS support is not there.
